async/delay_test.ts

```diff
@@ -1,6 +1,17 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 import { delay } from "./delay.ts";
-import { assert, assertRejects } from "../testing/asserts.ts";
+import {
+  assert,
+  assertInstanceOf,
+  assertRejects,
+  assertStrictEquals,
+} from "../testing/asserts.ts";
+
+// https://dom.spec.whatwg.org/#interface-AbortSignal
+function assertIsDefaultAbortReason(reason: unknown) {
+  assertInstanceOf(reason, DOMException);
+  assertStrictEquals(reason.name, "AbortError");
+}
```

Comment:
I've updated the existing tests to stop checking for the custom `"message"` property string in [Deno's implementation](https://github.com/denoland/deno/blob/v1.34.3/ext/web/03_abort_signal.js#L83) of the default `DOMException`, and instead check that the value complies with the attributes described in the [spec](https://dom.spec.whatwg.org/#interface-AbortSignal).

Reply:
This makes sense to me

Reply:
I would leave that in there to be honest, don't change the exception message, add the reason onto a field on the exception instead.

Reply:
@justinmchase Do you mean adding `.reason` field to the existing `new DOMException("Delay was aborted.", "AbortError")` error object?

Reply:
Yeah I suppose I did.